compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2114,10 +2114,10 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                 if self.predicate_may_hold(&try_obligation) && impls_future {
                     if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(span) {
                         if snippet.ends_with('?') {
-                            err.span_suggestion(
-                                span,
-                                "consider using `.await` here",
-                                format!("{}.await?", snippet.trim_end_matches('?')),
+                            err.span_suggestion_verbose(
+                                span.with_hi(span.hi() - BytePos(1)).shrink_to_hi(),
```

Comment:
This is the second occurrence of this pattern... but I have no idea how to generalize it in a useful way...

Reply:
We could add a method `Span::shrink_by_one(Self)` to avoid the `BytePos` play, but two calls would still be needed.

Reply:
yea, that's one solution, but I don't like it unless we have more such `- BytePos(1)` in rustc or clippy.

Reply:
We have a few in `rustc`, but I try to avoid them like the plague* whenever feasible.

* Really need to stop using that phrase, some people embrace the plague.